docs/cli/ilab-rag-retrieval.md

```diff
@@ -0,0 +1,221 @@
+# Design Proposal - Embedding Ingestion Pipeline And RAG-Based Chat
+
+**Author**: Daniele Martinoli
+
+**Version**: 0.1
+
+## 1. Introduction
+This document proposes enhancements to the `ilab` CLI to support workflows utilizing Retrieval-Augmented Generation
+(RAG) artifacts within `InstructLab`. The proposed changes introduce new commands and options for the embedding ingestion
+and RAG-based chat pipelines:
+* A new embedding ingestion command to process customer documentation, generate embeddings, and ingest them into a configured vector store.
+* An option to enhance the chat pipeline by using the stored embeddings to augment the context of conversations, improving relevance and accuracy.
+
+## 2. Proposed Commands
+### 2.1 Working Assumption
+This proposal aims to serve as a reference design to develop a Proof of Concept for RAG workflows, while
+also laying the foundation for future implementations of state-of-the-art RAG artifacts tailored to specific use
+cases.
+
+To minimize impact on current and future users, the following boundaries are defined:
+* Existing commands will only be updated to add flags or configurations that are maintainable in future versions.
+  * Configuration parameters will be marked as `Optional` to not affect existing runtimes.
+* New commands will be added using flags (e.g., `--abc`) or environment variables (e.g., `ILAB_ABC`) to avoid any configuration changes. 
+  This approach ensures configuration compatibility even if the commands are deprecated later.
+
+### 2.2 Embedding Ingestion Pipeline Command
+The proposal is to add a `process`  command to the `data` command group, with an explicit `--rag` flag to trigger
+the executions:
+```
+ilab data process --rag /path/to/docs/folder
+```
+
+The rationale behind this choice is that the `data process` command can support future workflows, making its
+introduction an investment to anticipate other needs.
+
+Since the RAG behavior is the only functionality of this new command, executions without the `--rag` option will result
+in no output for now.
+
+#### Command Purpose
+Generate the embeddings from the documents at */path/to/docs/folder* folder and store them in the
+configured vector database. These embeddings are intended to be used as augmented context in a RAG-based chat pipeline.
+
+#### Assumptions
+The provided documents must be in JSON format according to the InstructLab schema: this is the schema generated
+when transforming knowledge documents with the `ilab data generate` command (see 
+[Getting Started with Knowledge Contributions][1]).
+
+To simplify the execution of the transformation step, we introduce a `--transform` option which also includes the documents
+transformation, leveraging on the `instructlab-sdg` modules. 
+
+### Why We Need It
+This command streamlines the `ilab data generate` pipeline and eliminates the requirement to define a `qna` document,
```

Comment:
This is a really far-reaching design decision that could have a lot of consequences for the product. Looks like this came out of a meeting with the engine runtime team, was that recorded?

Reply:
recording link shared on Slack channel